lib/services/awsAdapter.js

```diff
@@ -37,13 +46,74 @@ function AWSAdapter (ec2) {
 					Value : instance.id
 				} ]
 			};
-			ec2.createTagsAsync(tags);
+			return ec2.createTagsAsync(tags);
 		})
 		.then(function () {
 			result.instanceID = instanceID;
+			self.beginPolling(result, instances, serverLog);
 			return result;
 		});
 	});
+
+	this.getPublicIPAddress = function (instanceID) {
+		return this.describeInstance(instanceID)
+		.then(function (instance) {
+			return	instance[ 0 ].PublicIpAddress;
+		});
+	};
+
+	this.describeInstance = function (instanceID) {
+		var params = {
+			InstanceIds : [ instanceID ]
+		};
+		return ec2.describeInstancesAsync(params)
+		.then(function (data) {
+			var instanceInfo = [];
+			data.Reservations.forEach(function (reservation) {
+				reservation.Instances.forEach(function (instance) {
+					instanceInfo.push(instance);
+				});
+			});
+			return instanceInfo;
+		});
+	};
+
+	this.beginPolling = function (result, instances, serverLog) {
+		var self = this;
+		return sshAdapter.startSshPolling(result.instanceID)
+		.then(function () {
+			serverLog([ "info", "AWSAdapter", "startSshPolling" ], "Ssh Polling is successful");
+			return self.getPublicIPAddress(result.instanceID);
+		})
+		.then(function (ipAddress) {
+			var updatedInstance;
+
+			updatedInstance = new Instance({
+				id    : result.id,
+				type  : result.type,
+				ami   : result.ami,
+				state : "ready",
+				uri   : "https://" + ipAddress,
+			});
+			instances.updateInstance(updatedInstance);
```

Comment:
Ok sorry I was looking at the error handling in the catch block. If updateInstance fails in the then block, it should ideally be caught in the catch block. I shall verify this.